libs/service-portal/access-control/delegations/src/components/access/AccessForm.tsx

```diff
@@ -224,6 +235,28 @@ export const AccessForm = ({ delegation, validityPeriod }: AccessFormProps) => {
         loading={updateLoading}
         error={updateError}
       />
+      <AccessDeleteModal
+        id={`access-delete-modal-${delegation?.id}`}
+        onDelete={() => {
+          history.push(
+            delegation.to?.nationalId === userInfo?.profile.nationalId
+              ? ServicePortalPath.AccessControlDelegationsToMe
+              : ServicePortalPath.AccessControlDelegations,
+          )
+        }}
+        onClose={() => setOpenDeleteModal(false)}
+        label={formatMessage(m.accessControl)}
+        title={formatMessage({
+          id: 'sp.settings-access-control:access-remove-modal-content',
+          defaultMessage: 'Ertu viss um að þú viljir eyða þessum aðgangi?',
+        })}
+        isVisible={openDeleteModal}
+        delegation={delegation as AuthCustomDelegation}
+        domain={{
+          name: delegation?.domain.displayName,
+          imgSrc: delegation?.domain.organisationLogoUrl,
+        }}
```

Comment:
You are passing the delegation to modal, so it could just read it from there instead.